volatility3/framework/symbols/windows/extensions/mft.py

```diff
@@ -21,3 +21,25 @@ def get_full_name(self) -> str:
             "string", encoding="utf16", max_length=self.NameLength * 2, errors="replace"
         )
         return output
+
+
+class MFTAttribute(objects.StructType):
+    """This represents an MFT ATTRIBUTE"""
+
+    def get_resident_filename(self) -> str:
+        # To get the resident name, we jump to relative name offset and read name length * 2 bytes of data
+        layer = self._context.layers[self.vol.layer_name]
+        attr_name_offset = self.vol.offset + self.Attr_Header.NameOffset
+       
+        return self._context.layers[layer.name].read(
```

Comment:
I feel like it'd be possible to make this a cast to `UNICODE_STRING` with some tweaking to the structure, but I'm ok with this as is if you don't want to.

Reply:
This might end up throwing decoding exceptions which will need catching where it gets called.